xcp_d/utils/hcp2fmriprep.py

```diff
@@ -4,165 +4,309 @@
+
+    subject_dir_fmriprep = os.path.join(out_dir, sub_ent)
+    anat_dir_fmriprep = os.path.join(subject_dir_fmriprep, "anat")
+    func_dir_fmriprep = os.path.join(subject_dir_fmriprep, "func")
+    os.makedirs(anat_dir_fmriprep, exist_ok=True)
+    os.makedirs(func_dir_fmriprep, exist_ok=True)
+
+    # Get necessary files
+    csf_mask = pkgrf("xcp_d", f"/data/masks/{volspace_ent}_{res_ent}_label-CSF_mask.nii.gz")
+    wm_mask = pkgrf("xcp_d", f"/data/masks/{volspace_ent}_{res_ent}_label-WM_mask.nii.gz")
+
+    # A dictionary of mappings from HCP derivatives to fMRIPrep derivatives.
+    # Values will be lists, to allow one-to-many mappings.
+    copy_dictionary = {}
+
+    # Collect anatomical files to copy
+    anat_dir_orig += "/MNINonLinear/"
+    t1w_orig = os.path.join(anat_dir_orig, "T1w_restore.nii.gz")
+    t1w_fmriprep = os.path.join(
+        anat_dir_fmriprep,
+        f"{sub_ent}_{volspace_ent}_{res_ent}_desc-preproc_T1w.nii.gz",
+    )
+    copy_dictionary[t1w_orig] = [t1w_fmriprep]
+
+    brainmask_orig = os.path.join(anat_dir_orig, "brainmask_fs.nii.gz")
+    brainmask_fmriprep = os.path.join(
+        anat_dir_fmriprep,
+        f"{sub_ent}_{volspace_ent}_{res_ent}_desc-brain_mask.nii.gz",
+    )
+    copy_dictionary[brainmask_orig] = [brainmask_fmriprep]
+
+    # NOTE: What is this file for?
+    ribbon_orig = os.path.join(anat_dir_orig, "ribbon.nii.gz")
+    ribbon_fmriprep = os.path.join(
+        anat_dir_fmriprep,
+        f"{sub_ent}_{volspace_ent}_{res_ent}_desc-ribbon_T1w.nii.gz",
+    )
+    copy_dictionary[ribbon_orig] = [ribbon_fmriprep]
+
+    dseg_orig = os.path.join(anat_dir_orig, "aparc.a2009s+aseg.nii.gz")
+    dseg_fmriprep = os.path.join(
+        anat_dir_fmriprep,
+        f"{sub_ent}_{volspace_ent}_{res_ent}_desc-aparcaseg_dseg.nii.gz",
+    )
+    copy_dictionary[dseg_orig] = [dseg_fmriprep]
+
+    # Create empty files to substitute as volume templates
+    affine = np.eye(4)
+    matrix = np.zeros(16)
+    nifti_file = nb.Nifti1Image(matrix, affine)
+    tempdir = tempfile.mkdtemp()
+    fake_nifti_1 = os.path.join(tempdir, "fakenifti1.nii.gz")
+    fake_nifti_2 = os.path.join(tempdir, "fakenifti2.nii.gz")
+    nb.save(nifti_file, fake_nifti_1)
+    nb.save(nifti_file, fake_nifti_2)
+
+    # Grab transforms
+    t1w_to_template_orig = fake_nifti_1
+    t1w_to_template_fmriprep = os.path.join(
+        anat_dir_fmriprep,
+        f"{sub_ent}_from-T1w_to-{volspace}_mode-image_xfm.nii.gz",
+    )
+    copy_dictionary[t1w_to_template_orig] = [t1w_to_template_fmriprep]
+
+    template_to_t1w_orig = fake_nifti_2
+    template_to_t1w_fmriprep = os.path.join(
+        anat_dir_fmriprep,
+        f"{sub_ent}_from-{volspace}_to-T1w_mode-image_xfm.nii.gz",
+    )
+    copy_dictionary[template_to_t1w_orig] = [template_to_t1w_fmriprep]
+
+    # Grab surface morphometry files
+    anat_dir_orig = in_dir
+    fsaverage_dir_orig = os.path.join(anat_dir_orig, "MNINonLinear", "fsaverage_LR32k")
```

Comment:
Why revert it, only to add `MNINonLinear` back to it?

Reply:
I pushed what I was talking about in [87678dd](https://github.com/PennLINC/xcp_d/pull/714/commits/87678dd567a0864cc225d5d7b4f8077984c86957).